enterprise/backend/src/metabase_enterprise/workspaces/models/workspace.clj

```diff
@@ -44,6 +45,10 @@
                    (update-in acc [(:workspace_id e) (type->grouping (:type e))] (fnil conj []) e))
                  {}))))
 
+(t2/define-before-insert :model/Workspace
+  [workspace]
+  (merge {:database_details {}} workspace))
```

Comment:
Why not just have null details as the default?

That way it seems easier to check whether the details have been generated yet, if we need to in a query.

Is this a limitation of `mi/transform-encrypted-json`? Maybe we can decorate it a nil-to-nil combinator.